Shopify/argo-admin-template
# feat: add globals dotenv

https://github.com/Shopify/app-extension-libs/issues/859

The 🎩 instructions will test both this PR and the [accompanying argo-admin-cli PR](https://github.com/Shopify/argo-admin-cli/pull/10)

## 🎩 Instructions

1. Change `package.json`

    ```diff
    - "@shopify/argo-admin-cli": "latest",
    + "@shopify/argo-admin-cli": "git+https://github.com/Shopify/argo-admin-cli.git#globals-test"
    ```

1. Update dependencies `npm install`

### Test

1. Generate a new project `npm run generate -- --type=SUBSCRIPTION_MANAGEMENT`

1. Create an environment file like `.envtest`. Add something like `ENDPOINT_API=https://www.example.com`.

1. Go to your generated `index.js|ts|tsx` file and add a console log to see your env variable is working.

    ```
    console.log('process.env.API_ENDPOINT', process.env.API_ENDPOINT);
    ```

1. Run local server `npm run server`. Visit `http://localhost:39351/` and open the Google Inspector Console for a console log message:

    <img width="532" alt="Screen Shot 2020-08-26 at 10 19 14 AM" src="https://user-images.githubusercontent.com/1744278/91315410-a6016f80-e785-11ea-9dc3-0e2c0d1ddf9d.png">

1. Run build `npm run build` and make sure the built file exists `build/main.js`
